pyoverkiz/models.py

```diff
@@ -271,6 +271,38 @@ def __init__(
         self.value = value
         self.type = DataType(type)
 
+    @property
+    def value_as_int(self) -> int | None:
+        if self.type == DataType.NONE:
+            return None
+        if self.type == DataType.INTEGER:
+            return cast(int, self.value)
+        raise TypeError(f"{self.name} is not an integer")
+
+    @property
+    def value_as_float(self) -> float | None:
+        if self.type == DataType.NONE:
+            return None
+        if self.type == DataType.FLOAT:
+            return cast(float, self.value)
+        raise TypeError(f"{self.name} is not a float")
+
+    @property
+    def value_as_bool(self) -> bool | None:
+        if self.type == DataType.NONE:
+            return None
+        if self.type == DataType.BOOLEAN:
+            return bool(self.value)
```

Comment:
Why cast it everywhere and not here?

Reply:
The issue is where I cast and not here.

It’s because of the returned type of value:

```
pyoverkiz/models.py:282: error: Argument 1 to "int" has incompatible type "Union[str, float, Dict[Any, Any], List[Any], None]"; expected "Union[str, bytes, SupportsInt, SupportsIndex, SupportsTrunc]"  [arg-type]

pyoverkiz/models.py:290: error: Argument 1 to "float" has incompatible type "Union[str, float, Dict[Any, Any], List[Any], None]"; expected "Union[SupportsFloat, SupportsIndex, str, bytes]"  [arg-type]
```

Reply:
But shouldn't we just use cast() everywhere? (since we should not need to do a implicit cast via int() / str() again?)

Reply:
We won't do any additional cast after this. I don't get your concern.

Reply:
`cast()` is free and only used by type checking.
`bool()` has a (very small) cost.
If you are sure that it is already a boolean, then `cast` is cheaper.

Reply:
Thx @epenet I was not sure about the more pythonic way to do it.